packages/tools/src/stateManagement/segmentation/SegmentationStateManager.ts

```diff
@@ -151,56 +144,146 @@ export default class SegmentationStateManager {
   }
 
   /**
-   * Returns the segmentation representation with the given UID.
-   * @param segmentationRepresentationUID - The UID of the segmentation representation.
-   * @returns The segmentation representation object.
+   * Adds a segmentation representation to the specified viewport.
+   * @param {string} viewportId - The ID of the viewport.
+   * @param {string} segmentationId - The ID of the segmentation.
+   * @param {SegmentationRepresentations} type - The type of segmentation representation.
+   * @param {RenderingConfig} renderingConfig - The rendering configuration for the segmentation.
    */
-  getSegmentationRepresentation(
-    segmentationRepresentationUID: string
-  ): SegmentationRepresentation | undefined {
-    return this.state.representations[segmentationRepresentationUID];
+  addSegmentationRepresentation(
+    viewportId: string,
+    segmentationId: string,
+    type: SegmentationRepresentations,
+    renderingConfig: RenderingConfig
+  ): void {
+    const enabledElement = getEnabledElementByViewportId(viewportId);
+
+    if (!enabledElement) {
+      return;
+    }
+
+    if (!this.state.viewports[viewportId]) {
+      this.state.viewports[viewportId] = [];
```

Comment:
Should the state updates be done in a way which doesn't modify the underlying state data?  That would mean you could use a frozen version of state safely for returning in the get state, and can in fact use an internal copy that is frozen and simply gets updated/re-frozen on changes.

Reply:
I like it, like React, done